### PR TITLE
rush-parallel 0.5.0 (new formula)

### PR DESCRIPTION
**Pull request by apcamargo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/rush-parallel.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  homepage "https://github.com/shenwei356/rush"
+  url "https://github.com/shenwei356/rush/archive/refs/tags/v0.4.2.tar.gz"
+  sha256 "801b17e279406c1649b401a2214148d344f673d0a1c9eed1b417655c1fefd114"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GO111MODULE"] = "auto"
+    system "go", "build", "-o", "rush"
```
Comment (SMillerDev) [inline Formula/rush-parallel.rb:12]:
```suggestion
    system "go", "build", *std_go_args(output: bin/"rush")
```

File: Formula/rush-parallel.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  homepage "https://github.com/shenwei356/rush"
+  url "https://github.com/shenwei356/rush/archive/refs/tags/v0.4.2.tar.gz"
+  sha256 "801b17e279406c1649b401a2214148d344f673d0a1c9eed1b417655c1fefd114"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "mod", "init", "rush"
+    system "go", "mod", "tidy"
```
Comment (SMillerDev) [inline Formula/rush-parallel.rb:12]: These should not be ran here, they should be fixed upstream.

**Review by SMillerDev — CHANGES_REQUESTED** [review]

File: Formula/rush-parallel.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(output: bin/"rush")
+  end
+
+  test do
+    assert_equal "01\n02\n03\n04\n",
+    shell_output("seq 1 4 | #{bin}/rush -k 'echo 0{}'")
```
Comment (carlocab) [inline Formula/rush-parallel.rb:16]:
```suggestion
    assert_equal <<~EOS, pipe_output("#{bin}/rush -k 'echo 0{}'", (1..4).to_a.join("\n"))
      01
      02
      03
      04
    EOS
```

**Review by SMillerDev — APPROVED** [review]

**Comment by apcamargo** [comment]
Thanks @SMillerDev! Do I need to do something to get this PR merged?

**Pull request closed without merging**